Always set lockdown_mode_byte, hardened_byte in pas_mte_config.c

Achierius opened the pull request:
#### 051e1cb9aeab209d5be4f734d8a8ce3d81e7b546
<pre>
Always set lockdown_mode_byte, hardened_byte in pas_mte_config.c
<a href="https://bugs.webkit.org/show_bug.cgi?id=301565">https://bugs.webkit.org/show_bug.cgi?id=301565</a>
<a href="https://rdar.apple.com/163557770">rdar://163557770</a>

Reviewed by Mark Lam.

Right now, there are certain code paths (privileged processes) that do
not ever set the lockdown mode byte or the hardened byte. This does not
have any effect at present because both of these bytes are only ever
used *within* WebContent to up the hardening level, but it’s confusing /
bug prone in the case that someone adds some condition that checks for
those bytes elsewhere in the code.

This patch fixes that by lifting the ldm-check and setting hardened_byte
in the privileged-process codepath.

Canonical link: <a href="https://commits.webkit.org/302741@main">https://commits.webkit.org/302741@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/db604d838cb9b957c8fb1585da89af66ca59b8be

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/129893 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/2154 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/40751 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/137286 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/81382 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/7dd66db4-4518-417b-bff5-0bbc1ac4f711) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/2107 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/2045 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/98947 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/66756 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/64a4e136-8954-4825-a8c4-2593fcd25f2e) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/132840 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/1588 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/116334 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/79636 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/7f941ae2-5660-4657-9064-feed3982b68e) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/1499 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/34463 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/80556 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/121885 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/110005 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/34968 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/139767 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/128345 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/1948 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/1812 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/107452 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/1993 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/112682 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/107337 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27356 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/1567 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/31156 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/54750 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/2021 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/161359 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/1835 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/40218 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/1870 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/1944 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

MenloDorian commented on Source/bmalloc/libpas/src/libpas/pas_mte_config.c:
```diff
@@ -98,6 +98,14 @@ static void pas_mte_do_initialization(void)
+    uint64_t ldmState = 0;
+    size_t sysCtlLen = sizeof(ldmState);
+    if (sysctlbyname("security.mac.lockdown_mode_state", &ldmState, &sysCtlLen, NULL, 0) >= 0 && ldmState == 1) {
+        wcp_is_hardened = true;
```
This setting of `wcp_is_hardened` does not make sense here.  `wcp_is_hardened` is a local variable in the `if (isWebContentProcess)` block below.

Achierius replied:
Fixed

MenloDorian commented on Source/bmalloc/libpas/src/libpas/pas_mte_config.c:
```diff
@@ -169,8 +169,10 @@ static void pas_mte_do_initialization(void)
-        *medium_byte = 1; // Tag libpas medium objects in privileged processes.
+    } else {
+        *medium_byte = 1; // Tag libpas medium objects in privileged processes
+        *hardened_byte = 0;
```
Shouldn't this be set to 1 because we want hardened mode by default?

Achierius replied:
Fixed

MenloDorian reviewed (approved):
r=me

Achierius committed:
Always set lockdown_mode_byte, hardened_byte in pas_mte_config.c
https://bugs.webkit.org/show_bug.cgi?id=301565
rdar://163557770

Reviewed by Mark Lam.

Right now, there are certain code paths (privileged processes) that do
not ever set the lockdown mode byte or the hardened byte. This does not
have any effect at present because both of these bytes are only ever
used *within* WebContent to up the hardening level, but it’s confusing /
bug prone in the case that someone adds some condition that checks for
those bytes elsewhere in the code.

This patch fixes that by lifting the ldm-check and setting hardened_byte
in the privileged-process codepath.

Canonical link: https://commits.webkit.org/302741@main
